src/UI/Components/Diagram/interfaces.ts

```diff
@@ -21,6 +21,7 @@ interface DictDialogData {
 interface Rule {
   name: string;
   type: TypeEnum;
+  attributeName?: string; //only used for inter-service relations
```

Comment:
If this is required for inter-service relations, isn't it worth having a type extending the default rule set? I'm no expert in TS, so maybe I'm wrong :D But it seems like this is what has been done so far in the codebase

Reply:
but the function that use that interface accept both embedded and inter-service relation, so there is no other place  that use that interface so that would be dubious in that case

Reply:
yes, I understand that, but it means that in the case of an inter-service relation, you have an optional where it would be a required field... Unless it's not required in the inter-service relation, then it's fine. But if it is required, then it biases your code, and enforces you to eliminate undefined cases, or cast values to bypass the linter.. 

Why not go for the kind:"InterserviceRelation" technique that is being used all over the rest of the app? It's just a suggestion

Reply:
I see that more like if there is that attributeName then it's inter-service_relation, if not then it's embedded, and it's simple enough.

Again I'm not strongly behind or against one of those solutions so I can adjust if you feel like current solution is unclear, I will take step back to look on the code, to check whether you proposition will improve readability, which could be the case, as I have biased POV at the moment as I stare at it for quite some time :)

Reply:
defining what kind of instance your dealing with, based on a field being present or not, is flakey and is not a clear solution that is maintainable. If someone new comes up in the team, and reads the code, they need explicit explanation to understand what's going on. That's something we don't want.